src/Analyzers/CSharp/CodeFixes/ConvertNamespace/ConvertNamespaceTransform.cs

```diff
@@ -186,53 +186,56 @@ public static (SourceText text, TextSpan semicolonSpan) ConvertNamespaceDeclarat
 
         private static FileScopedNamespaceDeclarationSyntax ConvertNamespaceDeclaration(NamespaceDeclarationSyntax namespaceDeclaration)
         {
-            var semiColon = SyntaxFactory.Token(SyntaxKind.SemicolonToken).WithoutTrivia();
+            // If the open-brace token has any special trivia, then move them to after the semicolon.
+            var semiColon = SyntaxFactory.Token(SyntaxKind.SemicolonToken)
+                .WithoutTrivia()
+                .WithTrailingTrivia(namespaceDeclaration.Name.GetTrailingTrivia())
+                .WithAppendedTrailingTrivia(namespaceDeclaration.OpenBraceToken.LeadingTrivia);
+
+            if (!namespaceDeclaration.OpenBraceToken.TrailingTrivia.All(static t => t.IsWhitespace()))
+                semiColon = semiColon.WithAppendedTrailingTrivia(namespaceDeclaration.OpenBraceToken.TrailingTrivia);
```

Comment:
general idea here is that we move the trivia on the `{` to the `;` (excluding just pure whitespace runs).  This has an added benefit of making sure that newlines around the `{` become the newlines of the `;` which feels intuitive/naturel in terms of the resultant namespace.